main.tf

```diff
@@ -1,28 +1,52 @@
-  bucket_versioning_enabled       = var.bucket_versioning_enabled ? "Enabled" : "Suspended"
-  cross_account_policy_name       = "${var.prefix}-cross-acct-policy-${random_id.uniq.hex}"
-  iam_role_arn                    = module.lacework_eks_audit_iam_role.arn
-  iam_role_external_id            = module.lacework_eks_audit_iam_role.external_id
-  cross_account_iam_role_name     = "${var.prefix}-ca-${random_id.uniq.hex}"
-  sns_name                        = "${var.prefix}${random_id.uniq.hex}"
-  firehose_iam_role_name          = "${var.prefix}-fh-${random_id.uniq.hex}"
-  firehose_policy_name            = "${var.prefix}-fh-policy-${random_id.uniq.hex}"
-  firehose_delivery_stream_name   = "${var.prefix}${random_id.uniq.hex}"
-  eks_cw_iam_role_name            = "${var.prefix}-cw-${random_id.uniq.hex}"
-  cw_iam_policy_name              = "${var.prefix}-cw-policy-${random_id.uniq.hex}"
-  cloudwatch_permission_resources = "arn:aws:logs:*:${data.aws_caller_identity.current.account_id}:log-group:/aws/eks/*:*"
-  log_regions                     = [for region in var.cloudwatch_regions : "logs.${region}.amazonaws.com"]
-  cluster_names                   = var.no_cw_subscription_filter ? [] : var.cluster_names
+  bucket_name                         = "${var.prefix}${random_id.uniq.hex}"
+  mfa_delete                          = var.bucket_versioning_enabled && var.bucket_enable_mfa_delete ? "Enabled" : "Disabled"
+  bucket_versioning_enabled           = var.bucket_versioning_enabled ? "Enabled" : "Suspended"
+  cross_account_policy_name           = "${var.prefix}-cross-acct-policy-${random_id.uniq.hex}"
+  iam_role_arn                        = module.lacework_eks_audit_iam_role.arn
+  iam_role_external_id                = module.lacework_eks_audit_iam_role.external_id
+  cross_account_iam_role_name         = "${var.prefix}-ca-${random_id.uniq.hex}"
+  sns_name                            = "${var.prefix}${random_id.uniq.hex}"
+  firehose_iam_role_name              = "${var.prefix}-fh-${random_id.uniq.hex}"
+  firehose_policy_name                = "${var.prefix}-fh-policy-${random_id.uniq.hex}"
+  firehose_delivery_stream_name       = "${var.prefix}${random_id.uniq.hex}"
+  eks_cw_iam_role_name                = "${var.prefix}-cw-${random_id.uniq.hex}"
+  cw_iam_policy_name                  = "${var.prefix}-cw-policy-${random_id.uniq.hex}"
+  cloudwatch_permission_resources     = "arn:aws:logs:*:${data.aws_caller_identity.current.account_id}:log-group:/aws/eks/*:*"
+  log_regions                         = [for region in var.cloudwatch_regions : "logs.${region}.amazonaws.com"]
+  cluster_names                       = var.no_cw_subscription_filter ? [] : var.cluster_names
+  create_kms_key                      = ((var.bucket_encryption_enabled && length(var.bucket_sse_key_arn) == 0) || (var.sns_topic_encryption_enabled && length(var.sns_topic_encryption_key_arn) == 0) || (var.kinesis_firehose_encryption_enabled && length(var.bucket_sse_key_arn) == 0)) ? 1 : 0
```

Comment:
It might be nice to add a comment explaining the logic here. It's hard to parse at a glance and would be helpful for others in the future.